[OneChat] Simple Chat UI

zacharyparikh opened the pull request:
## Summary

https://github.com/user-attachments/assets/47cf7b8e-fb43-43c8-b6b4-e7a379660d80


### Checklist

Check the PR satisfies following conditions. 

Reviewers should verify this PR satisfies this list as well.

- [ ] Any text added follows [EUI's writing guidelines](https://elastic.github.io/eui/#/guidelines/writing), uses sentence case text and includes [i18n support](https://github.com/elastic/kibana/blob/main/src/platform/packages/shared/kbn-i18n/README.md)
- [ ] [Documentation](https://www.elastic.co/guide/en/kibana/master/development-documentation.html) was added for features that require explanation or tutorials
- [ ] [Unit or functional tests](https://www.elastic.co/guide/en/kibana/master/development-tests.html) were updated or added to match the most common scenarios
- [ ] If a plugin configuration key changed, check if it needs to be allowlisted in the cloud and added to the [docker list](https://github.com/elastic/kibana/blob/main/src/dev/build/tasks/os_packages/docker_generator/resources/base/bin/kibana-docker)
- [ ] This was checked for breaking HTTP API changes, and any breaking changes have been approved by the breaking-change committee. The `release_note:breaking` label should be applied in these situations.
- [ ] [Flaky Test Runner](https://ci-stats.kibana.dev/trigger_flaky_test_runner/1) was used on any tests changed
- [ ] The PR  description includes the appropriate Release Notes section, and the correct `release_note:*` label is applied per the [guidelines](https://www.elastic.co/guide/en/kibana/master/contributing.html#kibana-release-notes-process)

### Identify risks

Does this PR introduce any risks? For example, consider risks like hard to test bugs, performance regression, potential of data loss.

Describe the risk, its severity, and mitigation for each identified risk. Invite stakeholders and evaluate how to proceed before merging.

- [ ] [See some risk examples](https://github.com/elastic/kibana/blob/main/RISK_MATRIX.mdx)
- [ ] ...

zacharyparikh committed:
WIP OneChat UI

elasticmachine commented:
<!-- buildkite-pr-bot-trigger-comment -->:robot: Jobs for this PR can be triggered through checkboxes. :construction: 

:information_source: To trigger the CI, please tick the checkbox below :point_down: 

- [ ] <!-- buildkite-pr-bot-checkbox-trigger:kibana-pull-request --> Click to trigger **kibana-pull-request** for this PR!
- [ ] <!-- buildkite-pr-bot-checkbox-trigger:kibana-deploy-project-from-pr --> Click to trigger **kibana-deploy-project-from-pr** for this PR!
- [ ] <!-- buildkite-pr-bot-checkbox-trigger:kibana-deploy-cloud-from-pr --> Click to trigger **kibana-deploy-cloud-from-pr** for this PR!

zacharyparikh committed:
ConversationsService list - return array instead of object

zacharyparikh committed:
Adapt Chat Components to Onechat

zacharyparikh committed:
WIP Add OneChat to Serverless Navigation Tree

zacharyparikh committed:
Remove unused conversations_panel components

zacharyparikh committed:
Remove unused chat_conversation_progress components

Commit:
[jedr] fix use_kibana hook to include internal services

zacharyparikh committed:
Fix conversation routing

zacharyparikh committed:
Handle conversation events

zacharyparikh committed:
Remove duplicate OneChat link in navigation tree

zacharyparikh committed:
Add new conversation button

zacharyparikh committed:
WIP Use persisted conversation rounds

Commit:
[jedr] hacky way to show tool calls with result! Please refactor

zacharyparikh committed:
Revert demo only changes

zacharyparikh committed:
Modify Onechat Link in Navigation Tree

zacharyparikh committed:
Fix warnings

zacharyparikh committed:
Optimistically update conversation round for streamed assistant message chunks

zacharyparikh committed:
Update query data for new conversation id

zacharyparikh committed:
Merge branch 'main' into onechat/simple-ui

zacharyparikh committed:
Merge remote-tracking branch 'upstream/main' into onechat/simple-ui

zacharyparikh committed:
Add feature flag to disable OneChat page

kibanamachine committed:
[CI] Auto-commit changed files from 'ts-node .buildkite/pipeline-resource-definitions/scripts/fix-location-collection.ts'

kibanamachine committed:
[CI] Auto-commit changed files from 'node scripts/telemetry_check'

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/311721)
* Commit: ae6c28b8d428883ce94540711b8256db42f9e580

### Failed CI Steps
* [Linting](https://buildkite.com/elastic/kibana-pull-request/builds/311721#01979e8a-45d7-4826-9d4f-6d4735ee176a)

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"failing","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/311721","number":311721,"commit":"ae6c28b8d428883ce94540711b8256db42f9e580"}],"number":311721}
buildkite-pr-comment-->

jedrazb reviewed (commented):
Can you add a video with a demo to the PR description? It makes reviewing easier :)

zacharyparikh committed:
Merge remote-tracking branch 'upstream/main' into onechat/simple-ui

pgayvallet reviewed (commented):
Didn't look at all the UI components, as I assumed those will evolve quite fast, so I focused on the data structures / archi / events.

Overall looks fine. A few things requiring work ihmo:


- There is a few things that got unnecessarily copied from workchat and looks unused, those should be removed
- we need to adapt the i18n labels to use the `xpack.onechat` prefix
- The conversation events / the way we convert events to the ui-facing shape was copied from workchat, which needs to be changed as we just don't have the same structure anymore. We need to re-use the existing onechat conversation / conversation round types instead.

zacharyparikh committed:
Add instructions to enable Chat UI to README

zacharyparikh committed:
Addressing Pierre feedback
- i18n labels
- cleanup unused files and objects
- remove unused hooks
- remove uses of workchat error types
- remove and adapt conversation event functions
- rename RoundTabAnswer component

zacharyparikh committed:
Remove unused query keys

zacharyparikh commented on x-pack/platform/plugins/shared/onechat/public/application/components/chat/i18n.ts:
```diff
@@ -0,0 +1,29 @@
+      defaultMessage: 'Healthy',
+    }),
+  },
+};
```
Addressed

zacharyparikh commented on x-pack/platform/plugins/shared/onechat/common/chat_events.ts:
```diff
@@ -0,0 +1,172 @@
+    callId: string;
+    result: string;
+  };
+}
```
Deleted

zacharyparikh commented on x-pack/platform/plugins/shared/onechat/common/conversation_events.ts:
```diff
@@ -0,0 +1,150 @@
+   */
+  createdAt: string;
+}
+
```
Deleted

zacharyparikh commented on x-pack/platform/plugins/shared/onechat/common/errors.ts:
```diff
@@ -0,0 +1,40 @@
+    super(code, message, meta);
+  }
+}
+
```
Addressed. Updated `use_chat.ts` to only send `OneChatError` instances to `onError` handler

zacharyparikh commented on x-pack/platform/plugins/shared/onechat/common/http_api/connectors.ts:
```diff
@@ -0,0 +1,12 @@
+
+export interface ListConnectorsResponse {
+  connectors: InferenceConnector[];
+}
```
Deleted

zacharyparikh commented on x-pack/platform/plugins/shared/onechat/public/application/components/chat/conversation/round_tab_sources.tsx:
```diff
@@ -0,0 +1,85 @@
+    return undefined;
+  }
+
+  const references = assistantMessage.citations;
```
Deleted this component and renamed `RoundTabAnswer` to `RoundAnswer`

zacharyparikh commented:
@pgayvallet Ready for re-review!

kibanamachine committed:
[CI] Auto-commit changed files from 'ts-node .buildkite/pipeline-resource-definitions/scripts/fix-location-collection.ts'

zacharyparikh committed:
Fix newConversationLabel property access in chat_header.tsx

zacharyparikh committed:
Fix i18n ids in server/plugin.ts

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/312637)
* Commit: 1a7a368b90369937059456eda3c3d0cdb2010eae

### Failed CI Steps
* [FTR Configs #113](https://buildkite.com/elastic/kibana-pull-request/builds/312637#0197a754-4b87-41fd-98b8-36faf236bdce)
* [FTR Configs #113](https://buildkite.com/elastic/kibana-pull-request/builds/312637#0197a782-c338-4f0b-959a-a9a493e1800c)

### Test Failures
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/312637#0197a754-4b87-41fd-98b8-36faf236bdce) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/312637/jobs/0197a754-4b87-41fd-98b8-36faf236bdce/artifacts/0197a782-97a3-499f-b986-b7808ee1d7cd) FTR Configs #<span></span>113 / Stack Management registers all UI Settings in the UsageStats interface
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/312637#0197a782-c338-4f0b-959a-a9a493e1800c) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/312637/jobs/0197a782-c338-4f0b-959a-a9a493e1800c/artifacts/0197a78a-df5d-4248-a042-9db21eb58c00) FTR Configs #<span></span>113 / Stack Management registers all UI Settings in the UsageStats interface

### History
* :broken_heart: [Build #312404](https://buildkite.com/elastic/kibana-pull-request/builds/312404) failed 37f117b91fd1a637bc7d424bb56bc986a71a3832
* :broken_heart: [Build #312109](https://buildkite.com/elastic/kibana-pull-request/builds/312109) failed 2b805940cee0eff3250256b31e5ee42041ff47ff
* :broken_heart: [Build #312059](https://buildkite.com/elastic/kibana-pull-request/builds/312059) failed b0115e4a00cfdea644092df5a66a54943098dc67
* :broken_heart: [Build #311721](https://buildkite.com/elastic/kibana-pull-request/builds/311721) failed ae6c28b8d428883ce94540711b8256db42f9e580

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"failing","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/312637","number":312637,"commit":"1a7a368b90369937059456eda3c3d0cdb2010eae"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/312404","number":312404,"commit":"37f117b91fd1a637bc7d424bb56bc986a71a3832"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/312109","number":312109,"commit":"2b805940cee0eff3250256b31e5ee42041ff47ff"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/312059","number":312059,"commit":"b0115e4a00cfdea644092df5a66a54943098dc67"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/311721","number":311721,"commit":"ae6c28b8d428883ce94540711b8256db42f9e580"}],"number":312637}
buildkite-pr-comment-->

pgayvallet commented:
/ci

elasticmachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [Buildkite Build](https://buildkite.com/elastic/kibana-pull-request/builds/312751)
* Commit: 1a7a368b90369937059456eda3c3d0cdb2010eae

### Failed CI Steps
* [FTR Configs #112](https://buildkite.com/elastic/kibana-pull-request/builds/312751#0197a7f4-982f-4e5a-8ec8-176841e78fd5)
* [FTR Configs #112](https://buildkite.com/elastic/kibana-pull-request/builds/312751#0197a81c-4b17-464e-bbf6-149cec26967f)

### Test Failures
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/312751#0197a7f4-982f-4e5a-8ec8-176841e78fd5) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/312751/jobs/0197a7f4-982f-4e5a-8ec8-176841e78fd5/artifacts/0197a81c-34d8-4803-b17d-c5d5572122b0) FTR Configs #<span></span>112 / Stack Management registers all UI Settings in the UsageStats interface
* [[job]](https://buildkite.com/elastic/kibana-pull-request/builds/312751#0197a81c-4b17-464e-bbf6-149cec26967f) [[logs]](https://buildkite.com/organizations/elastic/pipelines/kibana-pull-request/builds/312751/jobs/0197a81c-4b17-464e-bbf6-149cec26967f/artifacts/0197a822-ce8b-4aaa-a318-7a6aa7adcfe5) FTR Configs #<span></span>112 / Stack Management registers all UI Settings in the UsageStats interface

### History
* :broken_heart: [Build #312637](https://buildkite.com/elastic/kibana-pull-request/builds/312637) failed 1a7a368b90369937059456eda3c3d0cdb2010eae
* :broken_heart: [Build #312404](https://buildkite.com/elastic/kibana-pull-request/builds/312404) failed 37f117b91fd1a637bc7d424bb56bc986a71a3832
* :broken_heart: [Build #312109](https://buildkite.com/elastic/kibana-pull-request/builds/312109) failed 2b805940cee0eff3250256b31e5ee42041ff47ff
* :broken_heart: [Build #312059](https://buildkite.com/elastic/kibana-pull-request/builds/312059) failed b0115e4a00cfdea644092df5a66a54943098dc67
* :broken_heart: [Build #311721](https://buildkite.com/elastic/kibana-pull-request/builds/311721) failed ae6c28b8d428883ce94540711b8256db42f9e580

<!--buildkite-pr-comment-kibana-pull-request
{"builds":[{"buildStatus":{"state":"failing","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/312751","number":312751,"commit":"1a7a368b90369937059456eda3c3d0cdb2010eae"},{"buildStatus":{"state":"failing","success":false,"hasRetries":true,"hasNonPreemptionRetries":true},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/312637","number":312637,"commit":"1a7a368b90369937059456eda3c3d0cdb2010eae"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/312404","number":312404,"commit":"37f117b91fd1a637bc7d424bb56bc986a71a3832"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/312109","number":312109,"commit":"2b805940cee0eff3250256b31e5ee42041ff47ff"},{"buildStatus":{"state":"failed","success":false,"hasRetries":true,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/312059","number":312059,"commit":"b0115e4a00cfdea644092df5a66a54943098dc67"},{"buildStatus":{"state":"failed","success":false,"hasRetries":false,"hasNonPreemptionRetries":false},"url":"https://buildkite.com/elastic/kibana-pull-request/builds/311721","number":311721,"commit":"ae6c28b8d428883ce94540711b8256db42f9e580"}],"number":312751}
buildkite-pr-comment-->

zacharyparikh committed:
Register onechat:chat:enabled ui setting

zacharyparikh committed:
Remove "onechat" segment from public URL path

zacharyparikh committed:
Merge remote-tracking branch 'upstream/main' into onechat/simple-ui

zacharyparikh committed:
Align plugin setup with #224330

kibanamachine committed:
[CI] Auto-commit changed files from 'ts-node .buildkite/pipeline-resource-definitions/scripts/fix-location-collection.ts'

pgayvallet reviewed (approved):
LGTM 👍

zacharyparikh committed:
Merge remote-tracking branch 'upstream/main' into onechat/simple-ui

zacharyparikh committed:
Remove initial message context

zacharyparikh committed:
organize imports mount.tsx

zacharyparikh committed:
Remove unused onechat:chat:enabled feature flag

zacharyparikh committed:
Use "conversations" naming rather than "chat"

joemcelroy commented on x-pack/platform/plugins/shared/onechat/README.md:
````diff
@@ -256,3 +256,11 @@ Configure Claude Desktop by adding this to its configuration:
+To enable the Chat UI located at `/app/chat/`, add the following to your Kibana config:
+
+```yaml
+uiSettings.overrides:
````
thanks for starting this!

zacharyparikh replied:
No problem, it was Oren's suggestion

joemcelroy commented on x-pack/platform/plugins/shared/onechat/public/application/components/conversations/conversation_input_form.tsx:
```diff
@@ -0,0 +1,98 @@
+  `;
+
+  const inputFlexItemClass = css`
+    max-width: 900px;
```
same here with hardcoded width

joemcelroy reviewed (approved):
Looks great! Feel free to merge now without changes and address them in another PR, for the sake of a passing build :)

zacharyparikh commented on x-pack/platform/plugins/shared/onechat/public/application/components/conversations/conversation.tsx:
```diff
@@ -0,0 +1,86 @@
+
+const conversationPanelClass = css`
+  min-height: 100%;
+  max-width: 850px;
```
Styling will likely be changing anyways in follow up
